File: forumClient/src/Dto/app/PostInfo.ts
import type { UserBasicInfo } from "../UserInfo"

export type Post = {
    postId: string,
    threadId: string,
    isFirstPost: boolean,
    createdByUserId: string,
    postContent: string,
    createdDate: Date,
    replyToPostId: string,
    postReported: boolean,
    reportReason: string,
    reportedByUserId: string
}

export type PostBasicInfo = {
    post: Post
}

export type PostFullInfo = {
    post: Post
    createdByUser: UserBasicInfo
}

export type CreatePostRequest = {
    threadId: string,
    isFirstPost: boolean,
    createdByUserId: string,
    postContent: string,
    replyToPostId: string
}

export type ReportPostRequest = {
    threadId: string,
    postId: string,
    reportReason: string,
    reportedByUserId: string,
}

export type PostSummary = {
    totalPosts: number
}

File: server/RestApiServer/Dto/App/PostInfo.cs
using RestApiServer.Db;
using RestApiServer.Db.Users;
using RestApiServer.Utils;

namespace RestApiServer.Dto.App
{
    public class PostBasicInfo
    {
        public required PostEntry Post { get; set; }
    }

    public class PostFullInfo
    {
        public required PostEntry Post { get; set; }
        public required UserBasicInfo CreatedByUser { get; set; }
    }

    public class PostSummary 
    {
        public required int TotalPosts { get; set; } = 0;
    }
}